lib/generators/react/component_generator.rb

```diff
@@ -146,23 +187,58 @@ def webpacker?
         defined?(Webpacker)
       end
 
-       def parse_attributes!
-         self.attributes = (attributes || []).map do |attr|
-           name = ''
-           type = ''
-           options = ''
-           options_regex = /(?<options>{.*})/
+      def parse_attributes!
+        self.attributes = (attributes || []).map do |attr|
+          name = ''
+          type = ''
+          args = ''
+          args_regex = /(?<args>{.*})/
+
+          name, type = attr.split(':')
+
+          if matchdata = args_regex.match(type)
+            args = matchdata[:args]
+            type = type.gsub(args_regex, '')
+          end
+
+          if options[:ts]
+            { :name => name, :type => ts_lookup(name, type, args), :union => union?(args) }
+          else
+            { :name => name, :type => lookup(type, args) }
+          end
+        end
+      end
+
+      def union?(args = '')
+        return args.to_s.gsub(/[{}]/, '').split(',').count > 1
```

Comment:
I was wondering how Rails manages to parse out it's command line generators with curly braces such as `rails g migration add_thing_to_table thing:string{10}`
Turns out it's very similar to what you've done here already:
https://github.com/rails/rails/blob/1a5381ff0cf04af68a50bd04f265b9b8199e37b4/railties/lib/rails/generators/generated_attribute.rb#L45
But yours is a little easier to read 👍